hazelcast/src/test/java/com/hazelcast/internal/serialization/impl/compact/CompactNullablePrimitiveInteroperabilityTest.java

```diff
@@ -135,6 +139,51 @@ public void testWritePrimitiveReadNullable() {
         assertReadAsNullable(serializedRecord);
     }
 
+    @Test
+    public void testWritePrimitiveReadNullableCustomSerializer() {
+        FixedSizeFieldsDTO fixedSizeFieldsDTO = createFixedSizeFieldsDTO();
+        SerializationService serializationService = createSerializationService(FixedSizeFieldsDTOSerializerReadingNullable::new);
+
+        Data data = serializationService.toData(fixedSizeFieldsDTO);
+        FixedSizeFieldsDTO obj = serializationService.toObject(data);
+
+        assertEquals(obj, fixedSizeFieldsDTO);
+    }
+
+    @Test
+    public void testWriteArraysOfPrimitiveReadNullableCustomSerializer() {
+        ArrayOfFixedSizeFieldsDTO arrayOfFixedSizeFieldsDTO = createArrayOfFixedSizeFieldsDTO();
+        SerializationService serializationService2 = createSerializationService(ArrayOfFixedSizeFieldsDTOSerializerReadingNullable::new);
```

Comment:
we don't need the `2` at the end anymore. Similar for others (the data below, the other tests etc.)